src/fparser/two/utils.py

```diff
@@ -1500,64 +1594,74 @@ def tostr(self):
+    :type types: type or tuple of types
     :param int indent: extent to which to indent debug output.
     :param bool debug: whether or not to write textual representation of AST \
                        to stdout.
     :returns: a list of nodes
     :rtype: `list` of :py:class:`fparser.two.utils.Base`
     '''
     local_list = []
-    for child in children:
+
+    if not isinstance(node_list, (list, tuple)):
+        node_list = [node_list]
+
+    for child in node_list:
         if debug:
             if isinstance(child, str):
                 print(indent*"  " + "child type = ", type(child), repr(child))
             else:
                 print(indent*"  " + "child type = ", type(child))
-        if my_types is None or type(child) in my_types:
+        if types is None or isinstance(child, types):
             local_list.append(child)
-
-        # Depending on their level in the tree produced by fparser2003,
-        # some nodes have children listed in .content and some have them
-        # listed under .items. If a node has neither then it has no
-        # children.
-        if hasattr(child, "content"):
-            local_list += walk_ast(child.content, my_types, indent+1, debug)
-        elif hasattr(child, "items"):
-            local_list += walk_ast(child.items, my_types, indent+1, debug)
+        # Recurse down
+        local_list += walk(get_children(child), types, indent+1, debug)
 
     return local_list
 
 
-def get_child(root_node, node_type):
+def get_children(node):
     '''
-    Searches for the first immediate child of root_node that is of the
-    specified type.
+    Return a list containing the immediate children of the supplied node if
+    it is a sub-class of Base, otherwise return an empty list.
 
-    :param root_node: the parent of the child nodes we will search through.
-    :type root_node: :py:class:`fparser.two.utils.Base`
-    :param type node_type: the class of child node to search for.
+    :param node: the node for which to return the immediate children.
+    :type node: :py:class:`fparser.two.utils.Base` or str or NoneType
+
+    :returns: if the supplied node is a subclass of Base then return the \
+              list of its immediate children, otherwise return an empty list.
+    :rtype: list of :py:class:`fparser.two.utils.Base` or str
 
-    :returns: the first child node of type node_type that is encountered or \
-              None.
-    :rtype: :py:class:`fparser.two.utils.Base`
     '''
-    children = []
-    if hasattr(root_node, "content"):
-        children = root_node.content
-    elif hasattr(root_node, "items"):
-        children = root_node.items
-    for node in children:
-        if isinstance(node, node_type):
-            return node
-    return None
+    if not isinstance(node, Base):
+        return []
+    child_list = getattr(node, 'content', None)
+    if child_list is None:
+        child_list = getattr(node, 'items', [])
+    return child_list
+
+
+def get_parent(node):
```

Comment:
As discussed, please remove this in favour of a parent property/attribute in the `Base` base class.

Reply:
Done.